doc/experiment.rst

```diff
@@ -1,327 +1,440 @@
-in the Orchestrator, the keys by which they do this will overlap and they can end up accessing each
-others' data inadvertently. To prevent this situation, the SmartSim Entity object supports key
-prefixing, which automatically prepends the name of the model to the keys by which it is accessed.
-With this enabled, key overlapping is no longer an issue and ensemble members can use the same code.
-
-Under the hood, calling ensemble.enable_key_prefixing() causes the SSKEYOUT environment variable to
-be set, which in turn causes all keys generated by an ensemble member to be prefixed with its model
-name. Similarly, if the model for the ensemble member has incoming entities (such as those set via
-model.register_incoming_entity() or ensemble.register_incoming_entity()), the SSKEYIN environment
-variable will be set and the keys associated with those inputs will be automatically prefixed. Note
-that entities must register themselves as this is not done by default.
-
-Finally, please note that while prefixing is enabled by default for tensors, datasets, and aggregated
-lists of datasets, a SmartRedis client must manually call Client.use_model_ensemble_prefix() to
-ensure that prefixes are used with models and scripts.
-
-We modify the example above to enable key prefixing as follows:
-
-.. code-block:: bash
-
-  import numpy as np
-  from smartsim import Experiment
-
-  exp = Experiment("Training-Run", launcher="slurm")
-
-  # setup ensemble parameter space
-  learning_rate = list(np.linspace(.01, .5))
-  train_params = {"LR": learning_rate}
+``Ensemble`` will create the underlying ``Model`` objects. Users may also
+manually create and append ``Model(s)`` to an ``Ensemble``.
+Lastly, the :ref:`Ensemble API<ensemble_api>` supports launching Machine Learning Models, TensorFlow
+scripts and functions at runtime to enable AI and ML within an Ensemble
+Workload.
+
+Ensemble Prefixing
+------------------
+If each of the ensemble members attempt to use the
+same code to access their respective data in the Orchestrator,
+the names used to reference data, models, and scripts will be identical,
+and without the use of SmartSim and SmartRedis helper methods, ensemble members
+will end up inadvertently accessing or overwriting each other’s data. To prevent
+this situation, the SmartSim ``Ensemble`` object supports
+key prefixing, which automatically prepends the name
+of the model to the keys by which it is accessed. With
+this enabled, collision is resolved and
+ensemble members can use the same code.
+
+For example, assume you have two models in the ``Ensemble`` object,
+named `bar_0` and `bar_1`. In the application code you
+use the function ``Client.put_tensor("foo")``. With
+ensemble key prefixing turned on, the `bar_0` and `bar_1` Model applications
+can access the tensor `"foo"` by name without overwriting or accessing the other
+ensemble member's `"foo"` tensor.
+
+Create a Ensemble
```

Comment:
Can we change to "Ensemble Creation" to have parallel structure with other heading?